src/ersip_dialog.erl

```diff
@@ -178,17 +180,23 @@ uas_create(Request, Response) ->
 uas_pass_response(ReqSipMsg, RespSipMsg, #dialog{state = confirmed} = Dialog) ->
     UpdatedResp = uas_update_response(ReqSipMsg, RespSipMsg),
     {Dialog, UpdatedResp};
-uas_pass_response(ReqSipMsg, RespSipMsg, #dialog{state = early} = Dialog) ->
+uas_pass_response(ReqSipMsg, RespSipMsg, #dialog{state = early, request_method = RequestMethod} = Dialog) ->
     %% Independent of the method, if a request outside of a dialog generates
     %% a non-2xx final response, any early dialogs created through
     %% provisional responses to that request are terminated.
     case ersip_sipmsg:status(RespSipMsg) of
         Status when Status >= 300 ->
             terminate_dialog;
         _ ->
-            State = state_by_response(RespSipMsg),
             UpdatedResp = uas_update_response(ReqSipMsg, RespSipMsg),
-            {Dialog#dialog{state = State}, UpdatedResp}
+            CSeq = ersip_sipmsg:get(cseq, RespSipMsg),
+            case ersip_hdr_cseq:method(CSeq) of
```

Comment:
I believe that ersip_sipmsg:method/1 works for responses and gets method from cseq.